src/expected_loglik.jl

```diff
@@ -0,0 +1,151 @@
+abstract type QuadratureMethod end
+struct DefaultQuadrature <: QuadratureMethod end
+struct Analytic <: QuadratureMethod end
+
+struct GaussHermite <: QuadratureMethod
+    n_points::Int
+end
+GaussHermite() = GaussHermite(20)
+
+struct MonteCarlo <: QuadratureMethod
+    n_samples::Int
+end
+MonteCarlo() = MonteCarlo(20)
+
+_default_quadrature(_) = GaussHermite()
+
+"""
+    expected_loglik(quadrature::QuadratureMethod, y::AbstractVector, q_f::AbstractVector{<:Normal}, lik)
+
+This function computes the expected log likelihood:
+
+```math
+    ∫ q(f) log p(y | f) df
+```
+where `p(y | f)` is the process likelihood.
+
+`q(f)` is an approximation to the latent function values `f` given by:
+```math
+    q(f) = ∫ p(f | u) q(u) du
+```
+where `q(u)` is the variational distribution over inducing points (see
+[`elbo`](@ref)). The marginal distributions of `q(f)` are given by `q_f`.
+
+`quadrature` determines which method is used to calculate the expected log
+likelihood - see [`elbo`](@ref) for more details.
+
+# Extended help
+
+`q(f)` is assumed to be an `MvNormal` distribution and `p(y | f)` is assumed to
+have independent marginals such that only the marginals of `q(f)` are required.
```

Comment:
NB this might not work for heteroskedastic likelihoods (where we do care about correlations of different outputs for the same features).